### DISCUSSION TITLE
Client error

### DISCUSSION THREAD

**Discussion by AceOfSettlers** [fedorovvl/tso_client#38] [Q&A] [answered]
![Capture](https://user-images.githubusercontent.com/114313240/192117849-c007e929-a7d3-467e-a15f-149ab84b6414.JPG)
hello @fedorovvl I got this error while starting client , 
"obsolete client
please clear your browser cache and try to login again." 

i cleared all cash , install client again ... 

Thank you in advance if you can help me

**Comment by azurpc** [comment]
Hello, just go to azurpc tso, download swf and copy in %temp%\tso_portable and all is done ;)

**Reply by AceOfSettlers** [reply]
Thank you @azurpc for yout help , done what you have told me but still got the same error

**Reply by azurpc** [reply]
https://azurpc.wixsite.com/azurpctso/forum-1/settlers/azurpc-tso-toolbox-1
Based on the excellent work of fedorowvl

**Reply by azurpc** [reply]
> Thank you @azurpc for yout help , done what you have told me but still got the same error

Use toolbox v2, it's automatized the process

**Reply by azurpc** [reply]
I've made a folder error, update tomorrow or later this night

**Comment by azurpc** [comment]
ok it's done
https://drive.google.com/file/d/1a6oT829FE1_TeiB69G7LAriLE1rsiHco/view?usp=sharing

https://azurpc.wixsite.com/azurpctso/forum-1/settlers/azurpc-tso-toolbox-1

**Reply by AceOfSettlers** [reply]
![image](https://user-images.githubusercontent.com/114313240/192118744-7a404090-dd2a-48d3-8a12-7159099a125a.png)
I got this @azurpc , but not the client yet

**Reply by azurpc** [reply]
CLIENT DEJA ACTIF...
Close all/reboot and try again

**Reply by AceOfSettlers** [reply]
same @azurpc , thanx for you help but nothing works 

deleted temporary files,  cleard the cache ... restart the computer 
 
will try tommorow

**Reply by azurpc** [reply]
I'm on the game contact me

**Reply by SurferMortimer** [reply]
[https://www.virustotal.com/gui/file/27179f5216ceed5f754b15bb9096831155ea45bdeb63052ac1e8fcb829cc3d03?nocache=1](url)
Virustotal reports some Trojans in the file

**Reply by Cdrwbw** [reply]
Wahou Two years after

**Comment by azurpc** [comment]
Resolved ;)

**Comment by AceOfSettlers** [comment]
Thank you @azurpc for your help, i installed the client in another PC and i got the same probleme , it's like CLIENT is running in backround , do you have any clue @fedorovvl  ?

**Comment by fedorovvl** [comment, marked as answer]
try to download last release, run it once, then download this swf https://sirris.tsomaps.com/client_hanzo.swf
rename to client.swf and replace existing at %localappdata%/temp/tso_portable
run client.exe with --skip flag

**Reply by AceOfSettlers** [reply]
Working thanx @fedorovvl ! i replaced the swf after the .exe created it in the temp directory !

don't know how to run client.exe with --skip flag ? can you explain it to me ?

thanx  @azurpc  too for your help

**Reply by AceOfSettlers** [reply]
@fedorovvl  will update the swf ? to be downloaded automatically ?

**Reply by BasilMas** [reply]
> Wotking thanx @fedorovvl ! i replaced the swf after the .exe created it in the temp directory !
> 
> don't know how to run client.exe with --skip flag ? can you explain it to me ?
> 
> thanx @azurpc too for your help

You need to download https://sirris.tsomaps.com/client_hanzo.swf, rename and replace it in %localappdata%/temp/tso_portable
Then, you need to create shortcut for client.exe, and add ' --skip' (without quotes) in Object window like this 

![image](https://user-images.githubusercontent.com/76757249/192632009-c28407a3-f08a-45bc-b852-802c0aa110b0.png)

and run the client via this shortcut only!!!!

**Reply by AceOfSettlers** [reply]
@BasilMas Where do i find Object ? 
![image](https://user-images.githubusercontent.com/114313240/192647646-ba42c13c-32ac-49c2-9f38-59c8bf9d2895.png)

**Reply by BasilMas** [reply]
![image](https://user-images.githubusercontent.com/76757249/192746535-399c4f4d-0c1d-43b1-be43-e621a3f6c02a.png)

Here

**Reply by sqylogin** [reply]
I did the following:
- Delete everything on %localappdata%/temp/tso_portable
- Downloaded Client.exe version 1.4.9
- Ran Client.exe and allowed it to download files
- Downloaded client_hamzo.swf from https://sirris.tsomaps.com/client_hanzo.swf (size of file is 9,477,383 bytes - 1 byte more than the client.swf downloaded by client.exe), renamed to client.swf and replaced the old client.swf on %localappdata%/temp/tso_portable
- Created shortcut with --skip flag
- Ran client.exe using shortcut.

Unfortunately, I still get the error: Client Outdated - Please delete your browser cache and try to log in again

**Reply by BasilMas** [reply]
This one [https://sirris.tsomaps.com/client_hanzo.sw](https://sirris.tsomaps.com/client_hanzo.swf) does not work.

You can download client.swf from AzurPC site https://azurpc.wixsite.com/azurpctso/forum-1/settlers/azurpc-tso-toolbox-1, replace it in portable version folder and run client via shortcut with --skip parameter

**Reply by sqylogin** [reply]
I confirm that client.swf from https://drive.google.com/file/d/1ekEGWU23sInkYH61pI6G5xyU7rUvSHrR/view works. Thanks, BasilMas.

**Comment by conwayvm** [comment]
Hi fedorovvl
My problem is slightly different..   I reinstalled the client and now it doesnt give me the skip option that I used to get so I get the following error 
https://prnt.sc/rA8PZCcP-Qa9  Its the file loading failed one

Many thanks in advance

**Reply by BasilMas** [reply]
> Hi fedorovvl My problem is slightly different.. I reinstalled the client and now it doesnt give me the skip option that I used to get so I get the following error https://prnt.sc/rA8PZCcP-Qa9 Its the file loading failed one
> 
> Many thanks in advance

You need to start client.exe without any parameters for 'tso_portable' folder creation in proper pathway. After getting message about cache error you need to close client, rewrite client.swf in 'tso_portable' folder and use --skip option for play

**Reply by azurpc** [reply]
it's could help... (bat file)
@echo off

REM *** CHERCHER SI CLIENT EST OUVERT ***
ECHO CLIENT OUVERT OU NON >>C:\TSO\TEMP\LOG.TXT

tasklist | find "client.exe" /C /i >C:\TSO\TEMP\client.txt

ECHO VARIABLE CLIENTOK >>C:\TSO\TEMP\LOG.TXT
set /p clientok=<"C:\TSO\TEMP\client.txt"

REM *** LANCEMENT AUTO ***

:LANCEMENT AUTO

ECHO TEST SI AUTORUN >>C:\TSO\TEMP\LOG.TXT

IF EXIST "C:\TSO\TEMP\autorun.txt" (
	if '%clientok%' NEQ '0' (
	goto FIN ) ELSE (
ECHO:
ECHO: >>C:\TSO\TEMP\LOG.TXT
ECHO  *** DEMMARAGE PROCEDURE CLIENT ***
ECHO  *** DEMMARAGE PROCEDURE CLIENT *** >>C:\TSO\TEMP\LOG.TXT
ECHO  COPIE DU SWF...
ECHO  COPIE DU SWF... >>C:\TSO\TEMP\LOG.TXT

copy >>C:\TSO\TEMP\LOG.TXT "C:\TSO\TOOLBOX\ARCHIVES\UBISOFT\The Settlers Online\client.swf" "%LOCALAPPDATA%\Temp\tso_portable\client.swf"

echo  LANCEMENT UBICOLLECT...
echo  LANCEMENT UBICOLLECT... SI PAS EN EXECUTION >>C:\TSO\TEMP\LOG.TXT

	cd "c:\TSO\TOOLBOX\ARCHIVES\UBICOLLECT"
	tasklist | find "Ubicollect.exe" /C /i > nul
	if ERRORLEVEL 0 (
	taskkill /IM Ubicollect.exe /T /F >nul
	)
	Start /min UBICOLLECT.exe

	tasklist | find "Ubicollect.exe" /C /i > nul
	if ERRORLEVEL 1 (
	GOTO CLIENT_NO_UBI
	)

echo  CHARGEMENT DU CLIENT...
echo  CHARGEMENT DU CLIENT... MAXIMIZED NOSWF AUTO COLLECT >>C:\TSO\TEMP\LOG.TXT

ECHO:
ECHO: >>C:\TSO\TEMP\LOG.TXT

	cd "C:\TSO\TOOLBOX\ARCHIVES\SIRRIS\"
	call "C:\TSO\TOOLBOX\ARCHIVES\SIRRIS\client.exe" --window maximized --skip --autologin --collect
	)
)
GOTO :eof

:CLIENT_NO_UBI

echo >c:\tso\TEMP\msgbox.vbs Msgbox "PROBLEME avec Ubicollect (Antivirus ?), le chargement se poursuit, repondre NON à la prochaine boite de message suivant le login"
cscript >nul c:\tso\TEMP\msgbox.vbs

echo  CHARGEMENT DU CLIENT...
echo  CHARGEMENT DU CLIENT... MAXIMIZED NOSWF >>C:\TSO\TEMP\LOG.TXT

ECHO:
ECHO: >>C:\TSO\TEMP\LOG.TXT

	cd "C:\TSO\TOOLBOX\ARCHIVES\SIRRIS\"
	call "C:\TSO\TOOLBOX\ARCHIVES\SIRRIS\client.exe" --window maximized --skip
	)
)
GOTO :eof

:FIN
ECHO:
ECHO: >>C:\TSO\TEMP\LOG.TXT
ECHO  *** CLIENT DEJA ACTIF ***
ECHO  *** CLIENT DEJA ACTIF *** >>C:\TSO\TEMP\LOG.TXT
ECHO:

**Reply by conwayvm** [reply]
> 
> You need to start client.exe without any parameters for 'tso_portable' folder creation in proper pathway. After getting message about cache error you need to close client, rewrite client.swf in 'tso_portable' folder and use --skip option for play

Hi BasilMas

I am no techie   I really appreciate your answer  BUT can you explain for a dummy how to start the client with no parametres  I am like a toddler learniong to walk IT needs to be step by step :)

**Reply by BasilMas** [reply]
[conwayvm]
you need to double (or once) click on client.exe which you download from this repository

**Reply by AceOfSettlers** [reply]
@BasilMas @conwayvm @sqylogin  hello, i don't know how to creat a shortcut with --skip , when i creat it it's an empty shortcut that do nothing , can anyone share how to do it , step by step , thanx in advance

**Reply by AceOfSettlers** [reply]
@azurpc @fedorovvl

**Reply by BasilMas** [reply]
> @BasilMas @conwayvm @sqylogin hello, i don't know how to creat a shortcut with --skip , when i creat it it's an empty shortcut that do nothing , can anyone share how to do it , step by step , thanx in advance

I guess  you need to google step by step instruction for shortcut creation! for ex, https://www.thewindowsclub.com/create-desktop-shortcut-windows-10#:~:text=Right-click%20on%20its%20.exe,of%20the%20program%20executable%20file

**Reply by AceOfSettlers** [reply]
@BasilMas it is not the problem in the creation of the shortcut but the "--skip" parameter ... i don't know how to do it when i rename the shortcut i got an error or in the desctop it's empty shortcut ... i don't know if I made myself understood ... 
![image](https://user-images.githubusercontent.com/114313240/193459865-b32d7f51-698d-4ad7-920f-1534bae1530d.png)
![image](https://user-images.githubusercontent.com/114313240/193460021-3f4efeb3-eced-4140-a9f1-76113b7bf677.png)
![image](https://user-images.githubusercontent.com/114313240/193460060-2d9e6f22-845b-4b27-abd6-8fb55d742d6b.png)
![image](https://user-images.githubusercontent.com/114313240/193460077-a9339ac8-0e0b-484b-b70b-acaf2a309c55.png)

**Reply by BasilMas** [reply]
> @BasilMas it is not the problem in the creation of the shortcut but the "--skip" parameter ... i don't know how to do it when i rename the shortcut i got an error or in the desctop it's empty shortcut ... i don't know if I made myself understood ... ![image](https://user-images.githubusercontent.com/114313240/193459865-b32d7f51-698d-4ad7-920f-1534bae1530d.png) ![image](https://user-images.githubusercontent.com/114313240/193460021-3f4efeb3-eced-4140-a9f1-76113b7bf677.png) ![image](https://user-images.githubusercontent.com/114313240/193460060-2d9e6f22-845b-4b27-abd6-8fb55d742d6b.png) ![image](https://user-images.githubusercontent.com/114313240/193460077-a9339ac8-0e0b-484b-b70b-acaf2a309c55.png)


Firstly you need to type the space after client.exe and before two minuses like this "...client.exe --skip"   
Secondly if you remove client.exe from 'download' folder you 'break' the pathway for shortcut. 
Your main error is the follows: you try to rename file client.exe with wrong way!! don't type 'raccourci' after '.exe' - you 'break' exe-file this way and get get an empty field to enter like your picture. You need to type smth like 'client_rac.exe' or rename shortcut, not exe-file!!! 

Finally, you can unzip this archive into client folder which contain client.exe and run the game via TSO_start.bat
[TSO_start.zip](https://github.com/fedorovvl/tso_client/files/9692961/TSO_start.zip)

**Reply by AceOfSettlers** [reply]
@BasilMas Thanx with your BAT file it's working

**Comment by Chumazoid** [comment]
Все равно не работает

**Reply by BasilMas** [reply]
Все работает, если внимательно читать и делать  все четко по инструкции

**Comment by Urvdmih** [comment]
Нашел интересный баг... Если пароль сложный и содержит в себе спецсимвол одинарной кавычки в начале и середине пароля, то при передаче пароля через параметр --password передается часть в кавычках, остальное отбрасывается. Если же использовать функцию "запомнить" в клиенте с таким паролем - то создается лог-файл с ошибкой (причем при повторном запуске, первый проходит успешно).
[LastError.log](https://github.com/fedorovvl/tso_client/files/10433714/LastError.log)
